lib/src/sysutils.cpp

```diff
@@ -0,0 +1,42 @@
+#include <cstdio>
+#include <string>
+#include <cassert>
+#include <stdexcept>
+
+#include <dlisio/sysutils.hpp>
+
+#ifdef _WIN32
+    #include <windows.h>
+#endif
+
+namespace dlisio {
+
+std::FILE* sysopen(const std::string& path) noexcept (false) {
+    std::FILE* file;
+
+#ifdef _WIN32
+    auto wpathlen = MultiByteToWideChar(CP_UTF8, 0, path.c_str(), -1, NULL, 0);
+
+    /* can in theory fail if path contains illegal characters, though it is
+     * unlikely in practice
+     */
+    if (wpathlen <= 0) {
+        const auto msg = "dlisio::sysopen: unable to count wpath length";
+        throw std::runtime_error(msg);
+    }
+
+    std::wstring wpath;
+    wpath.resize(wpathlen);
+    auto written =
+        MultiByteToWideChar(CP_UTF8, 0, path.c_str(), -1,
+                            const_cast<wchar_t*>(wpath.c_str()), wpathlen);
```

Comment:
Pretty sure you should use `wpath.data()` here [1] ? Or as Microsoft do: `path[0]` [2]

[1] https://en.cppreference.com/w/cpp/string/basic_string
[2] https://docs.microsoft.com/en-us/windows/win32/api/stringapiset/nf-stringapiset-multibytetowidechar#code-example